core/AssetManager/UIAssetFetcher/PluginUmdAssetFetcher.php

```diff
@@ -109,6 +113,38 @@ private function getAllPluginUmds()
         return $allPluginUmds;
     }
 
+    private function checkForMissingPluginDependencies($pluginsToLoadOnInit)
+    {
+        foreach ($pluginsToLoadOnInit as $pluginName) {
+            $pluginDependencies = self::getPluginDependencies($pluginName);
+
+            // if the plugin dependencies has a plugin that is in $this->plugins, but not in $pluginsToLoadOnInit,
+            // the dependency was set to load on demand, and we report an error since this should only happen
+            // during development.
+            //
+            // if it's not in either $this->plugins and not in $pluginsToLoadOnInit, then it's not activated
+            // and we ignore it.
+            if (!empty(array_diff($pluginDependencies, $pluginsToLoadOnInit))
+                && empty(array_diff($pluginDependencies, $this->plugins))
+            ) {
+                throw new \Exception("Missing plugin dependency: $pluginName requires plugins "
```

Comment:
$pluginName  should that be variable instead of string

Reply:
I'm not sure what you mean by this, can you provide a code suggestion to clarify?

Reply:
I mean 

```suggestion
                throw new \Exception("Missing plugin dependency: ".$pluginName." requires plugins "
```